### PR TITLE
Pull Request 8

### PR DESCRIPTION
**Pull request by talgsternberg**
The "show number of comments" feature is working !

### CODE REVIEW

File: portfolio/src/main/webapp/script.js
Comment (jcespejo) [inline portfolio/src/main/webapp/script.js]: I'm not seeing where this `titleElement` is defined.

File: portfolio/src/main/webapp/script.js
Comment (jcespejo) [inline portfolio/src/main/webapp/script.js]: You're reusing the `commentElement` variable (another one on line 58) I believe.

Also, not sure if this is part of the walkthrough, but I'm not a fan of it using `const` in this situation.  Usually `const` means the variable will not be modified, but the appendChild calls below says otherwise.  I would instead use `let` or `var`.

File: portfolio/src/main/webapp/index.html
Comment (jcespejo) [inline portfolio/src/main/webapp/index.html]: This button I think would need a `type="submit"` tag to tell the form that clicking the button submits the form?

**Review by jcespejo — COMMENTED** [review]
Java looks good.  HTML I think is missing that one piece that would prevent you from submitting new comments.

JS has some mixup with variables being used, which I think would prevent you from displaying the comments on your page.

File: portfolio/src/main/webapp/script.js
Comment (talgsternberg) [reply portfolio/src/main/webapp/script.js]: They used const in the examples/walkthroughs but I'll try it with let/var instead and see what happens!

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java
Comment (jcespejo) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java]: Unfortunately not a good idea to do this here.  As mentioned in the chat, `countEntities()` would query the database and doing this in a loop means you're querying the database multiple times.  As a rule of thumb, try to limit the number of times you query a database since each time you do will require the database server to perform some actions for you (plus the time required to send the request and wait for the response from the database).

An alternative approach would be to do something like:
```
int total_comments_count = results.countEntities();
for (...) {
  if (... || commentNum < total_comments_count) {
  }
}
```

Comment (jcespejo) [reply portfolio/src/main/java/com/google/sps/servlets/DataServlet.java]: Also you maybe don't need to check for the total_comments_count at all, since the `results.asIterable()` for loop would auto exit if it runs out of comments to pull.  Meaning commentNum will never be greater than countEntities.

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java
Comment (jcespejo) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java]: supernit:  `commentNum++;` or `comentNum += 1;`

File: portfolio/src/main/webapp/index.html
Comment (jcespejo) [inline portfolio/src/main/webapp/index.html]: `loadComments()` here would load comments on page load, before a user has had the chance to make a choice regarding how many comments they would like to see.  This behavior is fine and is pretty common to load a default set.

Only thing is the servlet (which Github listed first and I then reviewed first) sounded like userChoice was required and only defaults to 3 on error scenarios (eg. if user picked a non-numeric value). In the case here, a blank userChoice is a legit action since it's used to pull the default comments set.  Maybe a quick fix for this would be to rename userChoice to someting more generic like "comments_quantity", which could be used both by the user actually picking something, or by the system's default behavior.

File: portfolio/src/main/webapp/index.html
Comment (jcespejo) [inline portfolio/src/main/webapp/index.html]: This might be what's causing some of the issues you were having (including the one with undefined/blank comments).  In your servlet you have the "get an X number of comments" hooked up to your GET action, and you have your "post a new comment" action hooked up to your POST action.  Since this form here submits a POST request to the servlet, it would instead try to post a new comment, which would have been blank in this scenario since you don't have the boxes to fill in the comment.

Definitely try switching this to GET for starters.

Then maybe a good opportunity for improvement would be to add validations to your POST comment action.  eg. don't save into datastore if the comment text is missing, or even better if it returns an error and the page displays the error.

File: portfolio/src/main/webapp/index.html
Comment (jcespejo) [inline portfolio/src/main/webapp/index.html]: nit:  The servlet considers zero to be an invalid value (and would return 3 comments instead).  It probably shouldn't be an option here since you wouldn't want your users submitting invalid values to your server.  Not to mention your server is built to return something that's different from what the user is expecting (zero comments).  I would suggest adjusting this to only allow 1-15, or adjust the servlet to allow 0-15.

**Review by jcespejo — COMMENTED** [review]
The issues you mentioned I think might be because you have POST on your "get comments" HTML form, where it should have been a GET action instead.  Try to see if that helps, then I can re-review and approve once fixed.

Your Java servlet and javascript both look pretty good and I think should work as expected.

File: portfolio/src/main/webapp/index.html
Comment (talgsternberg) [inline portfolio/src/main/webapp/index.html]: Now, shouldn't loadComments() only happen after I click the submit number of comments button?

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java
Comment (talgsternberg) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:72]: why does it always add an "undefined" comment?

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java
Comment (talgsternberg) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:48]: for some reason userChoice is always 3

Comment (jcespejo) [reply portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:48]: This was resolved by adding the choice param to the fetch call

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java
Comment (jcespejo) [reply portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:72]: Resolved by changing POST to GET on the get comments form

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java
Comment (jcespejo) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:105]: nit:  spaces

**Review by jcespejo — APPROVED** [review]
Changes look good.  I believe from our chat most of the issues you mentioned have been resolved.  If not, please feel free to send another pull request with updated code.

**Pull request merged**
